fix ml inference parameters description (#9246)
opensearch-project/documentation-website: _ingest-pipelines/processors/ml-inference.md

|:--- | :--- | :--- | :--- |
| `model_id` | String | Required | The ID of the ML model used by the processor. |
| `function_name` | String    | Optional for externally hosted models<br/><br/>Required for local models | The function name of the ML model configured in the processor. For local models, valid values are `sparse_encoding`, `sparse_tokenize`, `text_embedding`, and `text_similarity`. For externally hosted models, valid value is `remote`. Default is `remote`. |

> | `model_config` | Object    | Optional   | Custom configuration options for the ML model. For remote models, if set, this overrides the default parameters in connectors. For local models, it can be added to model_input to override the model configuration set during registration.. For more information, see [The `model_config` object]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#the-model_config-object). |

Reviewer: "For local models, it can be added to model_input to override the model configuration set during registration" Does this mean we can't use `model_config` for local model ?

Author: we could add `model_config` for local model, but the `model_config` needs to add to `model_input`, here is the example in below: [model_config](https://opensearch.org/docs/latest/ingest-pipelines/processors/ml-inference/#example-local-model)

> | `model_config` | Object    | Optional   | Custom configuration options for the ML model. For remote models, if set, this overrides the default parameters in connectors. For local models, it can be added to model_input to override the model configuration set during registration.. For more information, see [The `model_config` object]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#the-model_config-object). |

Reviewer 2: ```suggestion
| `model_config` | Object    | Optional   | Custom configuration options for the ML model. For externally hosted models, if set, this configuration overrides the default parameters in connectors. For local models, you can add `model_config` to `model_input` to override the model configuration set during registration. For more information, see [The `model_config` object]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#the-model_config-object). |
```

> | `model_config` | Object    | Optional   | Custom configuration options for the ML model. For externally hosted models, if set, this configuration overrides the default parameters in connectors. For local models, you can add `model_config` to `model_input` to override the model configuration set during registration. For more information, see [The `model_config` object]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#the-model_config-object). |

Reviewer 3: ```suggestion
| `model_config` | Object    | Optional   | Custom configuration options for the ML model. For externally hosted models, if set, this configuration overrides the default connector parameters. For local models, you can add `model_config` to `model_input` to override the model configuration set during registration. For more information, see [The `model_config` object]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#the-model_config-object). |
```
